cocos/2d/assembler/sprite/simple.ts

```diff
@@ -84,60 +86,19 @@ export const simple: IAssembler = {
 
         const dataList: IRenderData[] = renderData.data;
         const node = sprite.node;
-
-        const data0 = dataList[0];
-        const data3 = dataList[1];
         const matrix = node.worldMatrix;
-        const a = matrix.m00; const b = matrix.m01;
-        const c = matrix.m04; const d = matrix.m05;
-
-        const justTranslate = a === 1 && b === 0 && c === 0 && d === 1;
-
-        const tx = matrix.m12; const ty = matrix.m13;
-        const vl = data0.x; const vr = data3.x;
-        const vb = data0.y; const vt = data3.y;
-
-        if (justTranslate) {
-            const vltx = vl + tx;
-            const vrtx = vr + tx;
-            const vbty = vb + ty;
-            const vtty = vt + ty;
-
-            // left bottom
-            vData[0] = vltx;
-            vData[1] = vbty;
-            // right bottom
-            vData[9] = vrtx;
-            vData[10] = vbty;
-            // left top
-            vData[18] = vltx;
-            vData[19] = vtty;
-            // right top
-            vData[27] = vrtx;
-            vData[28] = vtty;
-        } else {
-            const al = a * vl; const ar = a * vr;
-            const bl = b * vl; const br = b * vr;
-            const cb = c * vb; const ct = c * vt;
-            const db = d * vb; const dt = d * vt;
-
-            const cbtx = cb + tx;
-            const cttx = ct + tx;
-            const dbty = db + ty;
-            const dtty = dt + ty;
-
-            // left bottom
-            vData[0] = al + cbtx;
-            vData[1] = bl + dbty;
-            // right bottom
-            vData[9] = ar + cbtx;
-            vData[10] = br + dbty;
-            // left top
-            vData[18] = al + cttx;
-            vData[19] = bl + dtty;
-            // right top
-            vData[27] = ar + cttx;
-            vData[28] = br + dtty;
+
+        const stride = renderData.floatStride;
+
+        for (let i  = 0; i < dataList.length; i++) {
+            const curData = dataList[i];
+            const vec3_temp = new Vec3();
```

Comment:
don't create temporary vectors inside logic code, create it in script scope and reuse the same object every time

Reply:
pay attention to any object creation and new

Reply:
Modified.